Fix CSV importing on PHP 8

fjorgemota opened the pull request:
Previously, CSV importing on PHP 8 failed because the uploaded file was always detected as being empty. However, after #4770, this was "fixed", but the line count computed was way too big compared to the real number of lines inside the file. This fixes the total line count computed by using `iterator_count()` to count the number of lines inside the file.

Fixes #4725

### Changes proposed in this Pull Request

* Use a different way of computing the total line count, by using `iterator_count()` instead of `SplFileObject::key()`;

### Testing instructions

* Install Sensei;
* On the final page of the Setup Wizard (which you can see via the URL `/wp-admin/admin.php?page=sensei_setup_wizard&step=ready`), with the title "You're ready to start creating online courses!", click on the "Install a Sample course" button;
*  The file should be imported successfully;

renatho reviewed (dismissed):
It worked for me, but I couldn't reproduce the issue in `master`.

And I had a weird behavior that it took so long (in both branches), and just completed after more than 2.000 requests. I suspect it's something in my env, because I had some warnings in the logs about the xdebug.

> Xdebug: [Step Debug] Could not connect to debugging client. Tried: ::1:9000 (from HTTP_X_FORWARDED_FOR HTTP header), localhost:9000 (fallback through xdebug.client_host/xdebug.client_port) :-(

That said, since it fixed the issue in your env, and the change seems safe, I'm approving it! =)

Also, since I couldn't test it very well, if someone else from @Automattic/evergreen could test it too, it would be nice.

fjorgemota commented:
> It worked for me, but I couldn't reproduce the issue in master.

Hey, just to confirm: Are you testing on PHP 8.0?

I looked here, and I can reproduce the issue 100% of the time on PHP 8, but on PHP 7, it indeed doesn't occur.

renatho commented:
> Are you testing on PHP 8.0?

Yep! 🤷‍♂️

fjorgemota commented:
I reverted my original commit sent with this PR and used another approach to count the total number of lines inside the file, this time using `iterator_count()` instead of `SplFileObject::key()`.

I also updated the description of the PR to refer to the actual changes made by this PR, and requested a new review.

Now both the performance and the bug should be fixed both in PHP 7.0 and in PHP 8.0. =)

renatho reviewed (commented):
Good job and great documentation! I'm just not approving because I couldn't test well in my env w/ PHP 8.0.

I'm taking the opportunity to add Nexus as reviewers too since they also worked on the issue previously. 😉

m1r0 reviewed (approved):
Works on PHP 7.0 and 8.0.

@fjorgemota @renatho thanks for fixing this!
